mms/metrics_manager.py

```diff
@@ -57,39 +72,64 @@ def start(metrics_write_to, namespace, mutex):
         mutex: object
             Mutex to prevent double thread writing on same resource
         """
-        MetricsManager.metrics['error_metric'] = Metric('errors', mutex,
-                                                        namespace=namespace,
-                                                        aggregate_method='interval_sum',
-                                                        write_to=metrics_write_to)
-        MetricsManager.metrics['request_metric'] = Metric('requests', mutex,
-                                                          namespace=namespace,
-                                                          aggregate_method='interval_sum',
-                                                          write_to=metrics_write_to)
-        MetricsManager.metrics['cpu_metric'] = Metric('cpu', mutex,
-                                                      namespace=namespace,
+        for model_name, model_class in models.items():
+            MetricsManager.metrics['5XX_errors'] = Metric('5XX_errors', mutex,
+                                                                model_name=model_name,
+                                                                unit=MetricUnit['count'],
+                                                                aggregate_method='interval_sum',
+                                                                write_to=metrics_write_to)
+            MetricsManager.metrics['4XX_errors'] = Metric('4XX_errors', mutex,
+                                                                model_name=model_name,
+                                                                unit=MetricUnit['count'],
+                                                                aggregate_method='interval_sum',
+                                                                write_to=metrics_write_to)
+            MetricsManager.metrics['requests'] = Metric('requests', mutex,
+                                                              model_name=model_name,
+                                                              unit=MetricUnit['count'],
+                                                              aggregate_method='interval_sum',
+                                                              write_to=metrics_write_to)
+            MetricsManager.metrics['overall_latency'] = Metric('overall_latency', mutex,
+                                                                      model_name=model_name,
+                                                                      unit=MetricUnit['MB'],
+                                                                      aggregate_method='interval_average',
```

Comment:
interval_average (and total_average) methods mean we are reporting snapshot of average value for a time window (or cumulative). it will be impossible for users to aggregate these values across hosts or models in cloudwatch. we need to report these as CW 'Statistics', not as individual data points that represent the average.

also, since reporting as Statistics means CW can't calculate percentiles (e.g. TP99s), we will need to calculate those and report them in the same statistics object. we can do that as separate PR, as long as we get the statistics structure right here.

Reply:
If we want to migrate metrics entirely to CW, the concern is we will lose the usage of other reporting method. I'm not sure. @aaronmarkham @lupesko

Reply:
I think any dahsboard tool will need the full statistical data to calculate percentiles, etc. So I do not think we're preventing integration with future dashboard providers.
I'm supportive of the usage of CW Statistics object.